src/python/pants/core/util_rules/partitions.py

```diff
@@ -83,7 +83,13 @@ def single_partition(
 @frozen_after_init
 @dataclass(unsafe_hash=True)
 @runtime_ignore_subscripts
-class _SubPartitionBase(Generic[PartitionKeyT, PartitionElementT]):
+class _BatchBase(Generic[PartitionKeyT, PartitionElementT]):
+    """Base class for a collection of elements that should all be processed in a single process.
+
+    For example, a collection of strings pointing to files that should be linted in one process, or
+    a collection of field-sets pointing at tests that should all execute in the same process.
+    """
+
     tool_name: str
     elements: tuple[PartitionElementT, ...]
     key: PartitionKeyT
```

Comment:
Should this become `partition_key` so its unambiguous what this is?

Reply:
Updated!

Reply:
Oops, missed some spots...